src/ipfs.ts

```diff
@@ -69,3 +70,66 @@ export async function publishEdit(args: PublishEditProposalArgs): Promise<string
 
   return await Micro.runPromise(upload);
 }
+
+type PublishImageParams =
+  | {
+      blob: Blob;
+    }
+  | {
+      url: string;
+    };
+
+export async function uploadImage(params: PublishImageParams) {
+  const formData = new FormData();
+  let blob: Blob;
+  if ('blob' in params) {
+    blob = params.blob;
+  } else {
+    // fetch the image and upload it to IPFS
+    const response = await fetch(params.url);
+    blob = await response.blob();
+  }
+
+  formData.append('file', blob);
+
+  const buffer = Buffer.from(await blob.arrayBuffer());
+  let dimensions: { width: number; height: number } | undefined;
+  try {
+    dimensions = imageSize(buffer);
+  } catch (error) {}
```

Comment:
Do we know in what ways `imageSize` can throw errors?

Reply:
Yes, this test covers the scenario: https://github.com/graphprotocol/grc-20-ts/pull/43/files#diff-d3b85c1189a0602716394461db79206d48b4b7ad6d958a76fb7175bb04727732R118-R130